pandas/core/dtypes/cast.py

```diff
@@ -1018,6 +1019,80 @@ def soft_convert_objects(
     return values
 
 
+def convert_dtypes(
+    input_array,
+    convert_string: bool = True,
+    convert_integer: bool = True,
+    convert_boolean: bool = True,
+):
+    """
+        Convert objects to best possible type, and optionally,
+        to types supporting ``pd.NA``.
+
+        Parameters
+        ----------
+        input_array : ExtensionArray or PandasArray
+        convert_string : bool, default True
+            Whether object dtypes should be converted to ``StringDtype()``.
+        convert_integer : bool, default True
+            Whether, if possible, conversion can be done to integer extension types.
```

Comment:
The original reason that an option such as `convert_integer` was asked for (and I think its primary use case), was to control the conversion of numpy int dtype to nullable Int dtype (because eg you might want to keep dtypes that are already int, and only convert object dtype to nullable int dtype). 
This is no longer possible now with this option, I think?

Reply:
No, if you set `convert_integer==True`, anything that was numpy int dtype will remain so.   And even if you have `infer_objects==True`, and you had `convert_integer==False`, and you started with an `object` dtype of all numpy int, it will end up as numpy int:
```
>>> s=pd.Series([1,2,3], dtype=object)
>>> s
0    1
1    2
2    3
dtype: object
>>> s.convert_dtypes(infer_objects=True, convert_integer=False)
0    1
1    2
2    3
dtype: int64
```

Reply:
> No, if you set convert_integer==True, anything that was numpy int dtype will remain so. 

I don't think that is correct? 
With the latest version of your branch (of an hour ago), I get:

```
In [14]: s = pd.Series([1,2,3], dtype=int)                                                                                                                                                                         

In [15]: s.convert_dtypes(convert_integer=True)                                                                                                                                                                    
Out[15]: 
0    1
1    2
2    3
dtype: Int64
```

where the original numpy dtype is converted to a nullable Int64

Reply:
Oops, I got it backwards. I meant to say: "No, if you set `convert_integer==False`, anything that was numpy int dtype will remain so."

Reply:
> Not sure why you'd want to convert to nullable ints if convert_integer==False

Because I might want to only *not* convert columns that are already integer ("don't convert *from* integer", vs "don't convert *to* integer", you can interpret `convert_integer` in two ways ;)). 

Note that this was the original mentioned use case when this first came up (https://github.com/pandas-dev/pandas/pull/30929#discussion_r366056855), to have an option not change  *existing* integer types.

Reply:
So is the confusion here because the label `convert_integer` could mean "convert to integer" (as implemented) vs. "convert from integer" ?  Setting `convert_integer=False` means existing integer types won't change which provides the behavior you are looking for.

I could make this clearer in the documentation, i.e., explaining what the `convert_*` options do when they are set to `False`.  Or we could consider changing the argument names to be `convert_to_integer`, `convert_to_string`, `convert_to_boolean`, or a combination of both (name changes and doc changes).

My preference is to enhance the docs in the notes section.  Your thoughts, @jorisvandenbossche ?

Reply:
> Setting convert_integer=False means existing integer types won't change which provides the behavior you are looking for.

No, it's not (AFAIU) what I am describing above. This disables *all* conversion to nullable integers, not only for existing integer types. 
Trying to explain again: I want a combination of conversion of integers as object dtype to nullable dtype and preserve existing integer numpy types as numpy types. With what combination of options can I do this in the latest version of this PR? (as far as I understand, I cannot)

Reply:
>I want a combination of conversion of integers as object dtype to nullable dtype and preserve existing integer numpy types as numpy types. With what combination of options can I do this in the latest version of this PR? (as far as I understand, I cannot)

That's correct.  It's not possible to have objects that look like integers go to nullable dtype, and preserve the existing integer numpy types as numpy types.  My idea is that in a resulting DataFrame, you go "all in" on the nullable integer types, or you use numpy int types, but that you wouldn't have a mix of `Int64` and `np.dtype(int)` in the same DataFrame